myokit/formats/sbml/_importer.py

```diff
@@ -35,334 +36,1065 @@ def __init__(self):
+            'timeUnit': SBMLmodel.get('timeUnits'),
+            'volumeUnit': SBMLmodel.get('volumeUnits'),
+            'areaUnit': SBMLmodel.get('areaUnits'),
+            'lengthUnit': SBMLmodel.get('lengthUnits'),
+            'extentUnit': SBMLmodel.get('extentUnits'),
+        }
+        for unitId in units:
+            unit = units[unitId]
+            if unit in self.userUnitDict:
+                self.userUnitDict[unitId] = self.userUnitDict[unit]
+            elif unit in SBML2MyoKitUnitDict:
+                self.userUnitDict[unitId] = SBML2MyoKitUnitDict[unit]
+            else:
+                self.userUnitDict[unitId] = None
+
+        # Initialise parameter and species dictionary
+        self.paramAndSpeciesDict = dict()
+
+        # Add compartments to model
+        compDict = dict()
+        comps = self._getListOfCompartments(SBMLmodel)
+        if comps:
+            for comp in comps:
+                idx = comp.get('id')
+                if not idx:
+                    raise SBMLError(
+                        'The file does not adhere to SBML 3.2 standards.'
+                        ' No compartment ID provided.')
+                name = comp.get('name')
+                if not name:
+                    name = idx
+                size = comp.get('size')
+                unit = comp.get('units')
+                if unit:
+                    if unit in self.userUnitDict:
+                        unit = self.userUnitDict[unit]
+                    elif unit in SBML2MyoKitUnitDict:
+                        unit = SBML2MyoKitUnitDict[unit]
+                else:
+                    dim = comp.get('spatialDimensions')
+                    if dim:
+                        dim = float(dim)  # can be non-integer
+                    if dim == 3.0:
+                        unit = self.userUnitDict['volumeUnit']
+                    elif dim == 2.0:
+                        unit = self.userUnitDict['areaUnit']
+                    elif dim == 1.0:
+                        unit = self.userUnitDict['lengthUnit']
+                    else:
+                        unit = None
+
+                # Create compartment
+                compDict[idx] = model.add_component(
+                    self._convert_name(name))
+
+                # Add size parameter to compartment
+                var = compDict[idx].add_variable('size')
+                var.set_unit(unit)
+                var.set_rhs(size)
+
+                # save size in container for later assignments/reactions
+                self.paramAndSpeciesDict[idx] = var
+
+        name = self._convert_name('myokit')
+        if 'MyoKit' in compDict:
+            raise SBMLError(
+                'The compartment ID <MyoKit> is reserved in a myokit'
+                ' import.')
+        compDict['MyoKit'] = model.add_component(name)
+
+        # Add parameters to model
+        params = self._getListOfParameters(SBMLmodel)
```

Comment:
If you make the function return an emtpy list instead of `None`, these three lines can just be:
```
for param in self._get_list_of_parameters(sbml_model)
```
But it might be even nicer to have a method `self._parse_parameters(model)` ?